docs/src/references/c8y-configuration-management.md

```diff
@@ -0,0 +1,162 @@
+* If the file `/etc/tedge/c8y/c8y_configuration_plugin.toml`
+  is not found, empty, ill-formed or not-readable
+  then only `c8y_configuration_plugin.toml` is managed from the cloud.
+* If the file `/etc/tedge/c8y/c8y_configuration_plugin.toml` is ill-formed
+  or cannot be read then an error is logged, but the operation proceed
+  as if the file were empty.
+  
+The behavior of the `c8y_configuration_plugin` is also controlled
+by the configuration of thin-edge:
+
+* `tedge config get mqtt.bind_address`: the address of the local MQTT bus.
+* `tedge config get mqtt.port`: the TCP port of the local MQTT bus.
+* `tedge config get tmp.path`: the directory where the files are updated
+  before being copied atomically to their targets.
+* `tedge config get log.path`: the directory used for logging.
+
+## Usage
+
+```shell
+$ c8y_configuration_plugin
+Thin-edge device configuration management for Cumulocity
+
+USAGE:
+    c8y_configuration_plugin [OPTIONS] [SMARTREST_OPERATION]
+
+OPTIONS:
+        --config-file <CONFIG_FILE>    [default: /etc/tedge/c8y/c8y_configuration_plugin.toml]
+    -h, --help                         Print help information
+    -V, --version                      Print version information
+
+SMARTREST_OPERATION:
+    The SmartRest2 message received on `c8y/s/ds` and triggering either a download or an upload.
+    This operation is then processed in its enterity by the `c8y_configuration_plugin`,
+    notifying the Cumulocity tenant of its progress (messages `501`, `502` and `503`).
+    
+    If no SmartRest2 operation is provided, `c8y_configuration_plugin`
+    sends a `119` message to the cloud tenant with the list of managed configuration files.
+```
+
+## Logging
+
+The `c8y_configuration_plugin` reports progress and errors in the log file `/var/log/tedge/c8y_configuration_plugin.log`
+
+* All upload and download operation requests are logged, when received and when completed,
+  with one line per file.
+* All changes to the list of managed file is logged, one line per change.
+* All errors are reported with the operation context (upload or download? which file?).
+
+## Internals
+
+Points that still need to be addressed:
+
+* __CRITICAL__ When the list of configurations is sent to the Cumulocity?
+  * This question is related to how the `c8y_configuration_plugin` is launched.
+  * If launched by the mapper when a `524` or `526` message is received on `c8y/s/ds`,
+    then the `c8y_configuration_plugin` must be launched at least once to send a `119`.
+    * This can be done on install. Or when the file `c8y_configuration_plugin.toml` is changed.
+      This is utterly fragile.
+    * One can update the mapper to run `[init]` operations.
+      This is more consistent with the current design and will be useful for other operations
+      (for instance, log management).
+  * The `c8y_configuration_plugin` command can also run as a daemon that
+    sends a first `119` message, subscribe to `c8y/s/ds`,
```

Comment:
I'm a bit wary of such control messages. We've already dealt with messages getting lost when processes don't start in a specific order and all that. Just worried if this will open up another set of similar issues, even though I can't think of anything concrete at the moment if we use a QoS > 1 and persistence that's already enabled on Mosquitto.